src/Act/CLI.hs

```diff
@@ -244,49 +252,140 @@ decompile' solFile' cid solver' timeout debug' = do
           putStrLn (prettyAct s)
 
 
-hevm :: FilePath -> Maybe FilePath -> Maybe ByteString -> Maybe ByteString -> Solvers.Solver -> Maybe Integer -> Bool -> IO ()
-hevm actspec sol' code' initcode' solver' timeout debug' = do
+hevm :: FilePath -> Maybe FilePath -> Maybe FilePath -> Maybe ByteString -> Maybe ByteString -> Maybe FilePath -> Solvers.Solver -> Maybe Integer -> Bool -> IO ()
+hevm actspec sol' vy' code' initcode' sources' solver' timeout debug' = do
   let config = if debug' then debugActConfig else defaultActConfig
   cores <- liftM fromIntegral getNumProcessors
   specContents <- readFile actspec
+  inputsMap <- processSources
   proceed specContents (addBounds <$> compile specContents) $ \ (Act store contracts) -> do
-    cmap <- createContractMap contracts
+    (cmap, lmap) <- createContractMap contracts inputsMap
     res <- runEnv (Env config) $ Solvers.withSolvers solver' cores 1 (naturalFromInteger <$> timeout) $ \solvers ->
-      checkContracts solvers store cmap
+      checkContracts solvers store lmap cmap
     case res of
       Success _ -> pure ()
       Failure err -> prettyErrs "" err
   where
-    createContractMap :: [Contract] -> IO (Map Id (Contract, BS.ByteString, BS.ByteString))
-    createContractMap contracts = do
-      foldM (\cmap spec'@(Contract cnstr _) -> do
+
+    -- Creates maps of storage layout modes and bytecodes, for all contracts contained in the given Act specification
+    createContractMap :: [Contract] -> Map (Maybe Id) (LayoutMode, ByteString, ByteString) -> IO (Map Id (Contract, BS.ByteString, BS.ByteString), Map Id LayoutMode)
+    createContractMap contracts inputsMap | Map.keys inputsMap == [Nothing] =
+      -- Singleton inputsMap with Nothing as contract Id means that '--vy' was given
+      case contracts of
+        [spec'@(Contract cnstr _)] -> do
+          let cid =  _cname cnstr
+              (_, initcode'', runtimecode') = fromJust $ Map.lookup Nothing inputsMap
+          pure (Map.singleton cid (spec', initcode'', runtimecode'), Map.singleton cid VyperLayout)
+        _ -> do
+          render (yellow "Warning: " <> text "Multiple contracts in spec mapped to a single Vyper file" <> line)
```

Comment:
We could also just throw an error here. It's very unlikely that the Act source will have multiple specs that match the same bytecode.